pulp_ansible/tests/functional/api/collection/test_sync.py

```diff
@@ -82,3 +92,51 @@ def test_successive_syncs_repo_version(self):
         path = urlsplit(repo["_latest_version_href"]).path
         latest_repo_version = int(path.split("/")[-2])
         self.assertEqual(latest_repo_version, number_of_syncs, repo)
+
+    def test_mirror_sync(self):
+        """Sync multiple plugin into the same repo with mirror as `True`.
+
+        This test targets the following issue: 5167
+
+        * `<https://pulp.plan.io/issues/5167>`_
+
+        This test does the following:
+
+        1. Create a repo.
+        2. Create two remotes
+            a. Role remote
+            b. Collection remote
+        3. Sync the repo with Role remote.
+        4. Sync the repo with Collection remote with ``Mirror=True``.
+        5. Verify whether the content in the latest version of the repo
+           has only Collection content and Role content is deleted.
+        """
+        # Step 1
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo["_href"])
+
+        # Step 2
+        role_remote = self.client.post(ANSIBLE_REMOTE_PATH, gen_ansible_remote())
+        self.addCleanup(self.client.delete, role_remote["_href"])
+
+        collection_remote = self.client.post(
+            ANSIBLE_COLLECTION_REMOTE_PATH,
+            gen_ansible_remote(url=ANSIBLE_GALAXY_COLLECTION_URL, whitelist=COLLECTION_WHITELIST),
+        )
+        self.addCleanup(self.client.delete, collection_remote["_href"])
+
+        # Step 3
+        sync(self.cfg, role_remote, repo)
+        repo = self.client.get(repo["_href"])
+        self.assertIsNotNone(repo["_latest_version_href"])
+        self.assertDictEqual(get_added_content_summary(repo), ANSIBLE_FIXTURE_CONTENT_SUMMARY)
```

Comment:
assertEqual also verifies type. 

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

No need to update. FYI.